e2e/glustershd_test.go

```diff
@@ -188,3 +188,148 @@ func testGranularEntryHeal(t *testing.T, tc *testCluster) {
 	// delete volume
 	r.Nil(client.VolumeDelete(volname))
 }
+
+func testSplitBrainOperation(t *testing.T, tc *testCluster) {
+	r := require.New(t)
+
+	var brickPaths []string
+	for i := 1; i <= 2; i++ {
+		brickPath := testTempDir(t, "brick")
+		brickPaths = append(brickPaths, brickPath)
+	}
+
+	volname := formatVolName(t.Name())
+
+	// create 2x2 dist-rep volume
+	createReq := api.VolCreateReq{
+		Name: volname,
+		Subvols: []api.SubvolReq{
+			{
+				ReplicaCount: 2,
+				Type:         "replicate",
+				Bricks: []api.BrickReq{
+					{PeerID: tc.gds[0].PeerID(), Path: brickPaths[0]},
+					{PeerID: tc.gds[0].PeerID(), Path: brickPaths[1]},
+				},
+			},
+		},
+		Force: true,
+	}
+	_, err := client.VolumeCreate(createReq)
+	r.Nil(err)
+
+	r.Nil(client.VolumeStart(volname, false), "volume start failed")
+
+	var optionReq api.VolOptionReq
+	pidpath := path.Join(tc.gds[0].Rundir, "glustershd.pid")
+	optionReq.Options = map[string]string{"replicate.self-heal-daemon": "off"}
+	optionReq.Advanced = true
+	r.Nil(client.VolumeSet(volname, optionReq))
+	r.False(isProcessRunning(pidpath), "glustershd is still running")
+
+	if _, err := os.Lstat("/dev/fuse"); os.IsNotExist(err) {
+		t.Skip("skipping mount /dev/fuse unavailable")
+	}
+
+	mntPath := testTempDir(t, "mnt")
+	defer os.RemoveAll(mntPath)
+
+	host, _, _ := net.SplitHostPort(tc.gds[0].ClientAddress)
+	err = mountVolume(host, volname, mntPath)
+	r.Nil(err, fmt.Sprintf("mount failed: %s", err))
+	defer syscall.Unmount(mntPath, syscall.MNT_FORCE|syscall.MNT_DETACH)
+
+	f, err := os.Create(mntPath + "/file1.txt")
+	r.Nil(err, fmt.Sprintf("file creation failed: %s", err))
+	f.Close()
+
+	var prevKilledBrick string
+	getBricksStatus, err := client.BricksStatus(volname)
+	r.Nil(err, fmt.Sprintf("brick status operation failed: %s", err))
+
+	for brick := range getBricksStatus {
```

Comment:
This iterates through both the bricks and kills them. and doesn't prevKilledBrick get assigned to the last brick in the loop? As we have only two bricks we can have one brick killed here in this loop and the another will be killed in the next loop. If you have better implementation, its welcome.

Reply:
It won't iterate through both bricks as `break` will force to end the loop as soon as it finds one brick PeerID  matching.

Reply:
sorry, my bad missed the break at the end.